core/variant/variant.h

```diff
@@ -447,6 +451,11 @@ class Variant {
 
 	operator IPAddress() const;
 
+	template <typename T>
+	_FORCE_INLINE_ operator TypedArray<T>() const { return operator Array(); }
+	template <typename K, typename V>
+	_FORCE_INLINE_ operator TypedDictionary<K, V>() const { return operator Dictionary(); }
```

Comment:
Both of these types should probably get move semantics. That would _really_ speed up conversions here. Probably out of scope for this PR though.